ext/odbc/odbc.stub.php

```diff
@@ -4,193 +4,193 @@
+ * @param resource $odbc
  * @return resource|false
  * @alias odbc_exec
  */
-function odbc_do($connection_id, string $query) {}
+function odbc_do($odbc, string $query) {}
 
 #ifdef PHP_ODBC_HAVE_FETCH_HASH
-/** @param resource $result */
-function odbc_fetch_object($result, int $rownumber = -1): stdClass|false {}
+/** @param resource $statement */
+function odbc_fetch_object($statement, int $row = -1): stdClass|false {}
 
-/** @param resource $result */
-function odbc_fetch_array($result, int $rownumber = -1): array|false {}
+/** @param resource $statement */
+function odbc_fetch_array($statement, int $row = -1): array|false {}
 #endif
 
 /**
- * @param resource $result_id
- * @param array $result_array
+ * @param resource $statement
+ * @param array $array
  */
-function odbc_fetch_into($result_id, &$result_array, int $rownumber = 0): int|false {}
+function odbc_fetch_into($statement, &$array, int $row = 0): int|false {}
 
-/** @param resource $result_id */
-function odbc_fetch_row($result_id, ?int $row_number = null): bool {}
+/** @param resource $statement */
+function odbc_fetch_row($statement, ?int $row = null): bool {}
 
-/** @param resource $result_id */
-function odbc_result($result_id, string|int $field): string|bool|null {}
+/** @param resource $statement */
+function odbc_result($statement, string|int $field): string|bool|null {}
 
-/** @param resource $result_id */
-function odbc_result_all($result_id, string $format = ""): int|false {}
+/** @param resource $statement */
+function odbc_result_all($statement, string $format = ""): int|false {}
 
-/** @param resource $result_id */
-function odbc_free_result($result_id): bool {}
+/** @param resource $statement */
+function odbc_free_result($statement): bool {}
 
 /** @return resource|false */
 function odbc_connect(string $dsn, string $user, string $password, int $cursor_option = SQL_CUR_USE_DRIVER) {}
 
 /** @return resource|false */
 function odbc_pconnect(string $dsn, string $user, string $password, int $cursor_option = SQL_CUR_USE_DRIVER) {}
 
-/** @param resource $connection_id */
-function odbc_close($connection_id): void {}
+/** @param resource $odbc */
+function odbc_close($odbc): void {}
 
-/** @param resource $result_id */
-function odbc_num_rows($result_id): int {}
+/** @param resource $statement */
+function odbc_num_rows($statement): int {}
 
 #if !defined(HAVE_SOLID) && !defined(HAVE_SOLID_30)
-/** @param resource $result_id */
-function odbc_next_result($result_id): bool {}
+/** @param resource $statement */
+function odbc_next_result($statement): bool {}
 #endif
 
-/** @param resource $result_id */
-function odbc_num_fields($result_id): int {}
+/** @param resource $statement */
+function odbc_num_fields($statement): int {}
 
-/** @param resource $result_id */
-function odbc_field_name($result_id, int $field_number): string|false {}
+/** @param resource $statement */
+function odbc_field_name($statement, int $field_number): string|false {}
```

Comment:
Elsewhere we do not make a distinction between `$field_number` and `$field_name` and just use `$field` (or `$column` depending on DB) for both. Whether it's a number or a name (a lot of APIs also accept both) is determined by the type.

(If we want to keep the distinction here, we should `$field_number` -> `$field_num` to keep the abbreviation consistent.)

Reply:
I'm not sure here. Only `odbc_field_num()` accepts a name; all other `odbc_field_*()` functions  expect the number (which looks strange to me). (ah, `odbc_result()` accepts either) But since we have the type, just `$field` looks good to me.